mindsdb/integrations/handlers/influxdb_handler/influxdb_handler.py

```diff
@@ -0,0 +1,232 @@
+from typing import Optional
+from collections import OrderedDict
+
+import pandas as pd
+import duckdb
+import io
+import requests
+
+from mindsdb_sql import parse_sql
+from mindsdb.integrations.libs.base import DatabaseHandler
+from mindsdb.utilities.log import get_log
+from mindsdb_sql.parser.ast.base import ASTNode
+
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+
+log = get_log()
+
+
+class InfluxDBHandler(DatabaseHandler):
+    """
+    This handler handles connection and execution of the Airtable statements.
+    """
+
+    name = 'influxdb'
+    def __init__(self, name: str, connection_data: Optional[dict], **kwargs):
+        """
+        Initialize the handler.
+        Args:
+            name (str): name of particular handler instance
+            connection_data (dict): parameters for connecting to the database
+            **kwargs: arbitrary keyword arguments.
+        """
+        super().__init__(name)
+        self.parser = parse_sql
+        self.dialect = 'influxdb'
+        self.connection_data = connection_data
+        self.kwargs = kwargs
+
+        self.connection = None
+        self.is_connected = False
+
+    def __del__(self):
+        if self.is_connected is True:
+            self.disconnect()
+
+    def connect(self) -> StatusResponse:
+        """
+        Set up the connection required by the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+
+        if self.is_connected is True:
+            return self.connection
+        
+        url = f"{self.connection_data['influxdb_url']}/query"
+        params = {
+            ('db',f"{self.connection_data['influxdb_db_name']}"),
+            ('q',f"{self.connection_data['influxdb_query']}" )
+        }
+        headers = {
+            "Authorization": f"Token {self.connection_data['influxdb_token']}",
+            "Accept": "application/csv",
+        }
+        response = requests.request("GET",url,params=params,headers=headers)
+        
+        if response.status_code == 200:
+            self.connection = response
```

Comment:
Since we are using API calls instead of using db client this should be changed because as a response we will get the data here.

Reply:
I added the code to verify whether the API response is successful and passing it to native_query. Can you please suggest a method on how to handle the connect() in case of API?